src/main/java/startsteps/ECommerceShop/config/JwtAuthenticationFilter.java

```diff
@@ -53,7 +57,16 @@ protected void doFilterInternal(@NonNull HttpServletRequest request,
                 context.setAuthentication(authToken);
                 SecurityContextHolder.setContext(context);
             }
-            filterChain.doFilter(request, response);
+
+            if (hasAdminRole(userDetails)) {
```

Comment:
No 😄. This will bring in more issues. So here you are doing a check if the user has an admin role, and if he does not you set the status as forbidden. This means that any request where the user is not admin the response is forbidden. Maybe change to "hasRole('ROLE_ADMIN')"? But make sure the name of your role is also "ROLE_ADMIN"

Reply:
i fixed what you mentioned. The code is running and was successfully testing in Postman